### ISSUE TITLE
On sources' Browse Chant pages, truncate Volpiano at 20 characters

### ISSUE THREAD

**Issue by jacobdgm** [DDMAL/CantusDB#476]
Investigating (and possibly related to) #472, I notice that the Browse Chant page is practically unusable because it appears we are displaying the entire text of each chant, as well as the entire melody of the chant (on http://206.12.93.196/chants/?source=123672):

![Screen Shot 2022-12-14 at 2 19 28 PM](https://user-images.githubusercontent.com/58090591/207693469-85709af3-9e18-4c33-b6e7-22e3ab72dc7d.png)

**Comment by jacobdgm** [comment]
I'm guessing this is because we default to displaying chants' fulltexts if they don't have `incipit`s, and that chants in this manuscript don't have incipits (and would likely be automatically fixed if we implemented #361). It wouldn't solve the problem of displaying chants' full `volpiano` fields though: 
![Screen Shot 2022-12-14 at 2 23 51 PM](https://user-images.githubusercontent.com/58090591/207694890-c5ec92f7-460c-46e1-80dd-0cfd46fc513d.png)

**Comment by annamorphism** [comment]
Displaying the full text happens on OldCantus too, and normally is fine--this page is particularly egregious because the chant texts happen to be so long, but that's not usually the case (see below). OldCantus only has a melodic snippet though (first four syllables? I can't quite tell).
![image](https://user-images.githubusercontent.com/67451875/207699519-75825cf6-4c1c-498f-99d0-704113e30dd8.png)

**Comment by jacobdgm** [comment]
Edit: heh, Anna got to this just before I did. It does appear to be the first 20 characters of volpiano, though.

OldCantus appears to display the entire fulltext (in a font that's smaller than `<small>`, which helps slightly). But it also seems they display only the first 20 characters of the volpiano:
![Screen Shot 2022-12-14 at 2 49 19 PM](https://user-images.githubusercontent.com/58090591/207700586-9786f9e5-8dd4-4da7-b5e8-e7c1735a7736.png)

**Comment by jacobdgm** [comment]
Perhaps this would be getting too whimsical/overengineered, but one solution could be to display the incipit, as well as a slightly longer snippet of fulltext where, if you click on it (or perhaps if you mouse over it), it shows you the entire text of the chant. This wouldn't be too difficult to do with `<details>` and `<summary>` tags, or with a `title` attribute.

**Comment by annamorphism** [comment]
There is a piece of OldCantus that does this! (It's a bit clunky.) I'll see if I can find it...

**Comment by annamorphism** [comment]
aha! it's when searching [chants within a manuscript](https://cantusdatabase.org/searchms/123723?name=&genre=All&mode=&feast=&cid=&volpiano=All&combine_op=contains&t=crucem) on OldCantus. 
Has some display problems, but means the idea was tried out on the site at some point. On the other hand, it was never implemented elsewhere--did it just not happen, or was it decided that this was not actually a desirable feature?
![image](https://user-images.githubusercontent.com/67451875/207706779-13c15205-e573-4204-88c9-3e2a75633b3a.png)

**Comment by jacobdgm** [comment]
> did it just not happen, or was it decided that this was not actually a desirable feature?

Interesting! I don't recall ever hearing about this.

My best guess is that it was decided to standardize/harmonize the various views for searching chants (like, it seems arbitrary to me that these "Details" dropdowns should be implemented only in the search-within-manuscript view and not in other search views)

My suggestion was something less involved - the summary/details tags would be fully contained within a single cell. What's on OldCantus strikes me as an effective design, but I'm struggling to picture how I would implement a full extra row appearing between the other rows in the table.

**Comment by jacobdgm** [comment]
As far as I can tell, we can basically fix this by doing what OldCantus does - displaying the entire fulltext, but truncating the volpiano at 20 characters. Going to rename the issue to reflect this.

**Comment by jacobdgm** [comment]
This is almost a simple fix, except when we use the django builtin `truncate_chars` in the template, it adds an ellipsis which the volpiano font doesn't know what to do with.
